core-rust/state-manager/src/mempool/priority_mempool.rs

```diff
@@ -97,12 +98,26 @@ pub struct MempoolData {
     pub source: MempoolAddSource,
 }
 
+// There is only 1 [`MempoolData`] per unique [`MempoolTransaction`]
+impl Hash for MempoolData {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.transaction.hash(state);
+    }
+}
+
 #[derive(Debug, Clone, Eq)]
 pub struct MempoolTransaction {
     pub validated: Box<ValidatedNotarizedTransactionV1>,
     pub raw: RawNotarizedTransaction,
 }
 
+// We can uniquely identify the whole [`MempoolTransaction`] just by the [`RawNotarizedTransaction`] hash.
+impl Hash for MempoolTransaction {
```

Comment:
Ditto - this feels weird/scary to me, and relatively expensive needing to hash a payload.

We should use a key like `NotarizedTransactionHash` instead for hashmaps.